### PR TITLE
:sparkles: Implement message passing

### PR DESCRIPTION
**Pull request by anantnrg**
Implement message passing so that we can communicate between the config thread and the main thread

### CODE REVIEW

**Review by NyxAlexandra — COMMENTED** [review]
couldn't these functions take just the values needed from Lua and move the "getting value from Lua" to the callsite?

**Review by NyxAlexandra — CHANGES_REQUESTED** [review]
delegate to the members of `FocusTarget`: https://github.com/Smithay/smithay/blob/7e2e3c4975aae6ce8f83ac6f63205e244b6fda6e/anvil/src/focus.rs#L107

**Review by NyxAlexandra — CHANGES_REQUESTED** [review]
delegate to the members of `FocusTarget`: https://github.com/Smithay/smithay/blob/7e2e3c4975aae6ce8f83ac6f63205e244b6fda6e/anvil/src/focus.rs#L107

File: lua/strata/actions.lua [hunk 1]
```diff
@@ -16,4 +16,37 @@ function module.spawn(cmd)
 	return function() api.spawn(cmd) end
 end
 
+--- Switches to a workspace
+---@param id number
+---@return function
+function module.switch_to_ws(id)
```
[the remark this answers is not in the document]
Comment (anantnrg) [reply lua/strata/actions.lua:22]: Hmm... could be. I just copy-pasted what @calops had done. I'll try that.

**Pull request merged**